props/problem-set-timeline.checkpoint.md

```diff
@@ -0,0 +1,393 @@
+import TimelineEvent from "./TimelineEvent";
+
+const Timeline = () => {};
+
+export default Timeline;
+```
+
+</details>
+
+<details>
+
+<summary>
+    <code>src/components/Timeline.css</code>
+</summary>
+
+```css
+.timeline {
+  width: 30%;
+  margin: auto;
+  text-align: left;
+}
+```
+
+</details>
+
+<details>
+
+<summary>
+    <code>src/components/TimelineEvent.js</code>
+</summary>
+
+```js
+import "./TimelineEvent.css";
+import Timestamp from "./Timestamp";
+
+const TimelineEvent = () => {};
+
+export default TimelineEvent;
+```
+
+</details>
+
+<details>
+
+<summary>
+    <code>src/components/TimelineEvent.css</code>
+</summary>
+
+```css
+.timeline-event {
+  display: grid;
+  grid-template: 2rem auto 2rem / 1fr 1fr;
+  padding: 0.5rem;
+  border-bottom: 1px solid #e6ecf0;
+  background-color: #fff;
+}
+
+.timeline-event:hover {
+  background-color: #f5f8fa;
+}
+
+.event-person {
+  grid-area: 1 / 1 / span 1 / span 1;
+  margin-top: 0.5rem;
+  font-weight: bolder;
+}
+
+.event-status {
+  grid-area: 2 / 1 / span 1 / -1;
+}
+
+.event-time {
+  grid-area: 1 / 2 / span 1 / span 1;
+  margin-top: 0.5rem;
+  text-align: right;
+}
+```
+
+</details>
+
+<details>
+
+<summary>
+    <code>src/components/Timestamp.js</code>
+</summary>
+
+```js
+import moment from "moment";
+
+const Timestamp = (props) => {
+  const time = moment(props.time);
+  const absolute = time.format("MMMM Do YYYY, h:mm:ss a");
+  const relative = time.fromNow();
+
+  return <span title={absolute}>{relative}</span>;
+};
+
+export default Timestamp;
+```
+
+</details>
+
+## Feature Requirements
+
+In this assignment, modify the following three components:
+
+1. `App`
+1. `Timeline`
+1. `TimelineEvent`
+
+The `Timestamp` component is **_already fully implemented_**. You will _not_ need to modify it during this assignment.
+
+![Social media timeline web app. The top header reads "Ada Lovelace's social media feed." There are three different timeline events. Each timeline event has the timestamp "3 days ago." They also include a person's name and a status message.](../assets/props_problem-set-timeline_example.png)  
+_Fig. Social media timeline web app. The top header reads "Ada Lovelace's social media feed." There are three different timeline events. Each timeline event has the timestamp "3 days ago." They also include a person's name and a status message._
+
+### `App` Component
+
+The `App` component is a container component. It is responsible for:
+
+- Holding the data about the timeline, which contains all timeline events.
+- Rendering one instance of the `Timeline` component.
+- Passing the timeline data to the `Timeline` component.
+
+Notice that our modified `App` component imports a JSON object, and declares the object in a variable named `timelineData`:
+
+```js
+import timelineData from "./data/timeline.json";
+```
+
+### `Timeline` Component
+
+The `Timeline` component is also a container component. It's responsible for:
+
+- Reading the timeline data from `App`
```

Comment:
reading implies it must do something active to get it, but it's just getting passed in
```suggestion
- Receiving the timeline data from `App`
```